src/constraint/field.ts

```diff
@@ -38,7 +37,22 @@ export const FIELD_CONSTRAINTS: EncodingConstraintModel<FieldQuery>[] = [
     satisfy: (fieldQ: FieldQuery, _: Schema, __: PropIndex<Wildcard<any>>, ___: QueryConfig) => {
       return (fieldQ.field === '*') === (fieldQ.aggregate === 'count');
     }
-  // TODO: minCardinalityForBin
+  },{
+    name: 'minCardinalityForBin',
+    description: 'binned quantitative field should not have too low cardinality',
+    properties: [Property.BIN, Property.FIELD, Property.TYPE],
+    allowWildcardForProperties: false,
+    strict: true,
+    satisfy: (fieldQ: FieldQuery, schema: Schema, _: PropIndex<Wildcard<any>>, opt: QueryConfig) => {
+      if (fieldQ.bin && fieldQ.type === Type.QUANTITATIVE) {
+        // We remove bin so schema can infer the raw unbinned cardinality.
+        // Q: Is this OK, or should I instead duplicate fieldQ and use delete operator 
```

Comment:
This is fine.   If you want the whole fieldQ, you could do this (which is super short). 
```
const {bin: _, ...nonBinFieldQ} = fieldQ;
```